tests/test_odilo.py

```diff
@@ -0,0 +1,342 @@
+
+        acsm_data = self.sample_data("fulfill_ok_acsm.acsm")
+        self.api.queue_response(200, content=acsm_data)
+
+        self.fulfill('ACSM')
+
+    def test_22_fulfill_ebook_streaming(self):
+        checkout_data, checkout_json = self.sample_json("patron_checkouts.json")
+        self.api.queue_response(200, content=checkout_json)
+
+        self.licensepool.identifier.identifier = '00011055'
+        self.fulfill('EBOOK_STREAMING')
+
+    def fulfill(self, internal_format):
+        print('Fulfill test: ' + internal_format)
+
+        fulfillment_info = self.api.fulfill(self.PATRON, self.PIN, self.licensepool, internal_format)
+        ok_(fulfillment_info, msg='Cannot Fulfill !!')
+
+        if fulfillment_info.content_link:
+            print 'Fulfill link: %s' % fulfillment_info.content_link
+        if fulfillment_info.content:
+            print 'Fulfill content: %s' % fulfillment_info.content
+
+    #################
+    # Hold tests
+    #################
+
+    def test_31_already_on_hold(self):
+        already_on_hold_data, already_on_hold_json = self.sample_json("error_hold_already_in_hold.json")
+        self.api.queue_response(403, content=already_on_hold_json)
+
+        assert_raises(AlreadyOnHold, self.api.place_hold, self.PATRON, self.PIN, self.licensepool,
+                      'ejcepas@odilotid.es')
+
+        print 'Test hold already on hold ok!'
+
+    def test_32_place_hold(self):
+        print('Place hold test...')
+        hold_ok_data, hold_ok_json = self.sample_json("place_hold_ok.json")
+        self.api.queue_response(200, content=hold_ok_json)
+
+        hold_info = self.api.place_hold(self.PATRON, self.PIN, self.licensepool, 'ejcepas@odilotid.es')
+        ok_(hold_info, msg="HoldInfo null --> place hold failed!")
+        print 'Hold ok: %s' % hold_info.identifier
+
+    #################
+    # Patron Activity tests
+    #################
+
+    def test_41_patron_activity_invalid_patron(self):
+        patron_not_found_data, patron_not_found_json = self.sample_json("error_patron_not_found.json")
+        self.api.queue_response(404, content=patron_not_found_json)
+
+        assert_raises(PatronNotFoundOnRemote, self.api.patron_activity, self.PATRON, self.PIN)
+
+        print 'Test patron activity --> invalid partron ok!'
+
+    def test_42_patron_activity(self):
+        print ('Patron activity test...')
+        patron_checkouts_data, patron_checkouts_json = self.sample_json("patron_checkouts.json")
+        patron_holds_data, patron_holds_json = self.sample_json("patron_holds.json")
+        self.api.queue_response(200, content=patron_checkouts_json)
+        self.api.queue_response(200, content=patron_holds_json)
+
+        loans_and_holds = self.api.patron_activity(self.PATRON, self.PIN)
+        if loans_and_holds:
+            print 'Found: %i loans and holds' % len(loans_and_holds)
+        else:
+            print 'No loans or holds found'
```

Comment:
You know the outcome here -- either there are loans and holds or there aren't. The case you're expecting should be asserted.

Reply:
Right, I've already changed it.